api/v1alpha1/nutanix_clusterconfig_types.go

```diff
@@ -76,3 +84,39 @@ func (s NutanixPrismCentralEndpointSpec) ParseURL() (string, int32, error) {
 
 	return hostname, int32(port), nil
 }
+
+// NutanixFailureDomains is a list of FDs.
+type NutanixFailureDomains []NutanixFailureDomain
+
+// NutanixFailureDomain configures failure domain information for Nutanix.
+type NutanixFailureDomain struct {
```

Comment:
Is there some validation we want to do with PE Clusters and Subnets, like setting either this or those values there?

Reply:
capx handles it by checking fd set or not first. https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/blob/v1.4.0-alpha.2/controllers/nutanixmachine_controller.go#L834 need to see if we need to handle that in caren since caren does not have any notion of sequence of which patch will get called first. do you have any proposals?